network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/PhaseTapChangerImpl.java

```diff
@@ -148,4 +146,35 @@ public static void validateStep(TapChangerStepAttributes step, TapChangerParent
             throw new ValidationException(parent, "step alpha is not set");
         }
     }
+
+    // equals and hashcode are override because of ValidationUtil.checkPhaseTapChangerRegulation that
+    // remove the tap changer of the list of getAllTapChangers()
+    // and it does not work if equals and hascode check this.attributesGetter
```

Comment:
```suggestion
      // equals and hashCode are overridden to ensure correct behavior of the PhaseTapChanger 
      // in hash table-based collections (e.g., HashSet, HashMap). Without these overrides, the default 
      // implementations include this.attributesGetter, which can lead to incorrect behavior in 
      // hash-based collections by affecting instance identification, retrieval and removal.
```
Same for RatioTapChanger?